applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.h

```diff
@@ -151,6 +152,32 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) ElasticIsotropic3D
         return StressMeasure_Cauchy;
     }
 
+    /**
+     * @brief Returns the initial stress vector if it is defined in the geometry
+     */
+    const Vector GetInitialStressVector(Parameters& rParameterValues)
+    {
+        const auto &r_geometry = rParameterValues.GetElementGeometry();
+        if (r_geometry.Has(INITIAL_STRESS_VECTOR)) {
+            return r_geometry.GetValue(INITIAL_STRESS_VECTOR);
+        } else {
+            return ZeroVector(this->GetStrainSize());
+        }
+    }
+
+    /**
+     * @brief Returns the initial strain vector if it is defined in the geometry
+     */
+    const Vector GetInitialStrainVector(Parameters& rParameterValues)
+    {
+        const auto &r_geometry = rParameterValues.GetElementGeometry();
+        if (r_geometry.Has(INITIAL_STRAIN_VECTOR)) {
```

Comment:
hm so this is an additional `if` in each access

can you please make some performance tests as to how much this influences

Reply:
How do you do this test?

Reply:
That's why in my branch I created an entire class with the imposed deformations. The idea would be to add to the current CL a template , so derive by default from ConstitutiveLaw, and being possible to derive from ConstitutiveLawWithImposedDeformation. Then there will not be overhead

Reply:
This is a philosofical question, it is better to generalize the existing Linear elastic Law with this initial strains or stresses or to create a derived one with this new capability. In my opinion I think that the 1st option is the less confusing (also less quantity of basic CL) but the overhead is necessary right?

Reply:
> This is a philosofical question, it is better to generalize the existing Linear elastic Law with this initial strains or stresses or to create a derived one with this new capability. In my opinion I think that the 1st option is the less confusing (also less quantity of basic CL) but the overhead is necessary right?

It is necessary unless you separate the laws between the ones with initial strain and the ones without it as I said, which has it disadvantages

Reply:
What's your opinion @RiccardoRossi ?

Reply:
Even though I have a limited knowledge about the CL implementation, I support @philbucher 's and @loumalouomega 's observations as this implies introducing an if statement per Gauss pt.